src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -155,7 +156,7 @@ FilterDAGInfoPtr generateFilterActions(
 
     /// Using separate expression analyzer to prevent any possible alias injection
     auto syntax_result = TreeRewriter(context).analyzeSelect(query_ast, TreeRewriterResult({}, storage, storage_snapshot));
-    SelectQueryExpressionAnalyzer analyzer(query_ast, syntax_result, context, metadata_snapshot);
+    SelectQueryExpressionAnalyzer analyzer(query_ast, syntax_result, context, metadata_snapshot, {}, false, {}, prepared_sets);
```

Comment:
@KochetovNicolai I'm not 100% sure this is the right fix, but it seems it fixes the issue.

Reply:
Looks ok to me